Samples/src/Arithmetic/Adder Example.ipynb

```diff
@@ -0,0 +1,232 @@
+    "    Message(\"[Using the adder by Takahashi et al. (arXiv:0910.2530)]\");\n",
+    "    RippleCarryAdderTTK(qint1, qint2, carry);\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Running the adder\n",
+    "\n",
+    "We can now use this wrapper in our code as a subroutine. As an example, let's write a function that initializes two quantum integers to classical values (e.g., 4 and 5) before calling our `Add` operation."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "application/json": "[\"TestAddition\"]",
+      "text/html": [
+       "<ul><li>TestAddition</li></ul>"
+      ],
+      "text/plain": [
+       "TestAddition"
+      ]
+     },
+     "execution_count": 2,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "operation TestAddition() : Int {\n",
+    "    let firstNumber = 4;\n",
+    "    let secondNumber = 5;\n",
+    "    let numBits = 3;\n",
+    "\n",
+    "    mutable result = 0;\n",
+    "    Message($\"Calculating {firstNumber} + {secondNumber}...\");\n",
+    "    using ((register1, register2, carry) = (Qubit[numBits], Qubit[numBits], Qubit())) {\n",
+    "        let qint1 = LittleEndian(register1);\n",
+    "        let qint2 = LittleEndian(register2);\n",
+    "        \n",
+    "        // initialize the two quantum integers:\n",
+    "        InPlaceXorLE(firstNumber, qint1);\n",
+    "        InPlaceXorLE(secondNumber, qint2);\n",
+    "        \n",
+    "        // perform addition\n",
+    "        Add(qint1, qint2, carry);\n",
+    "        \n",
+    "        // and measure the result\n",
+    "        let mCarry = (MResetZ(carry) == One) ? 1 | 0;\n",
+    "        set result = 2^numBits * mCarry + MeasureInteger(qint2);\n",
+    "        ResetAll(register1 + register2);\n",
+    "    }\n",
+    "\n",
+    "    // Return the measured result\n",
+    "    return result;\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can now test our adder using the simulator:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Calculating 4 + 5...\n",
+      "[Using the adder by Takahashi et al. (arXiv:0910.2530)]\n"
+     ]
+    },
+    {
+     "data": {
+      "application/json": "9",
+      "text/plain": [
+       "9"
+      ]
+     },
+     "execution_count": 3,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "%simulate TestAddition "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Exchanging implementations\n",
+    "Furthermore, we can quickly exchange the underlying adder implementation as follows:"
```

Comment:
It might be good to similarly have a link to the Cuccaro *et al.* adder.

Reply:
Done